server/tserver/src/main/java/org/apache/accumulo/tserver/scan/ScanTask.java

```diff
@@ -73,30 +73,24 @@ public boolean cancel(boolean mayInterruptIfRunning) {
 
     if (state.compareAndSet(INITIAL, CANCELED)) {
       interruptFlag.set(true);
-      resultQueue = null;
+      resultFuture = null;
```

Comment:
Should this be either cancel() or complete() - setting it to null will block future access to the future - but I don't think it impacts the current task if it is running.  This happens two places in this file.